operators/pkg/controller/elasticsearch/driver/nodes.go

```diff
@@ -143,3 +108,26 @@ func (d *defaultDriver) reconcileNodeSpecs(
 	//  - grow and shrink
 	return results
 }
+
+func expectedResources(
```

Comment:
nit: I would not expect a function called `expectedResources` to have a side effect on the cluster

Reply:
I don't think there is a side effect on the cluster?
I guess what you refer to is that we patch the config of the expected resources for zen1/zen2, but those are just the expected resources in memory, we don't change anything in the apiserver here, nor make any ES request.